### PR TITLE
Implement btrfs resize support

### LINKED ISSUE #4083: Filesystem resize should support btrfs resize
### Summary

It would be nice if the filesystem module could resize btrfs filesystems.

I tried to get into the code of the module but my python skills a not good enough. 
Until now I resize the btrfs filesystem with

```bash (paste below)
btrfs filesystem resize max /mountpoint
```

mountpoint is calculated with:

```bash (paste below)
mount | sed -nr 's#^{{item.0.disk.dev}}{{item.1.num}} on (.+) type btrfs .*#\\1#gp' |head -1`
```

### Issue Type

Feature Idea

### Component Name

filesystem

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by elara-leitstellentechnik**
##### SUMMARY
Implement btrfs resize support

Fixes #4083 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
filesystem

##### ADDITIONAL INFORMATION
I tested this locally, to make sure it works. The comment on the other filesystems made me believe, that I cannot test this with the CI, since it requires the filesystem to be mounted. Is that right?

### CODE REVIEW

File: tests/integration/targets/filesystem/defaults/main.yml
Comment (felixfontein) [inline tests/integration/targets/filesystem/defaults/main.yml:19]: Have you tried setting `grow` to `True` and see what happens? Might be worth a try :)

**Review by felixfontein — COMMENTED** [review]
Thanks for your contriubtion! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)?

File: tests/integration/targets/filesystem/defaults/main.yml
Comment (elara-leitstellentechnik) [reply tests/integration/targets/filesystem/defaults/main.yml:19]: As suspected, it doesn't work:
```
TASK [filesystem : Expand filesystem] *********************************************************************************************************************************************
fatal: [testhost]: FAILED! => {"changed": false, "msg": "/tmp/ansible.lal4h9yf.test/img needs to be mounted for Btrfs operations"}
```

File: changelogs/fragments/4465-btrfs-resize.yml
Comment (felixfontein) [inline changelogs/fragments/4465-btrfs-resize.yml:3]:
```suggestion
  - filesystem - add support for resizing btrfs (https://github.com/ansible-collections/community.general/issues/4465).
```

File: tests/integration/targets/filesystem/defaults/main.yml
Comment (felixfontein) [reply tests/integration/targets/filesystem/defaults/main.yml:19]: Good question. If you're interested in trying to change that, feel free to create a new PR for that once this is merged. (Doing that in this PR is probably a bit too much.)

**Review by felixfontein — APPROVED** [review]
Will merge in ~one week if nobody else comments by then.

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@elara-leitstellentechnik thanks for contributing this!
@russoz thanks for reviewing!

**Pull request merged**
